### DISCUSSION TITLE
How do I determine if a property has a backing field?

### DISCUSSION THREAD

**Discussion by dylanenabled** [LemonAppDev/konsist#792] [Ask a Question]
### Wanted end result.

I have a test which ensures that kotlinx `@Serializable` classes use snake case `@SerialName` for the fields that will be serialised (the actual names use camel case). However some of the fields are properties that are computed via lazy delegates and so are not serialized, thus they don't need the `@SerialName` annotation and the test should skip them.

I want to be able to check in my test if the property has a backing field or not to skip those properties.

### What you tried so far.

I am currently using a check for `hasImplementation || delegateName != null` to skip but it seems possible that even if hasImplementation is true, it could still have a backing field.

**Comment by igorwojda** [comment]
Thx for the feedback.

We will add a dedicated `hasBackingField` property:
https://lemonappdev.atlassian.net/browse/KON-600

In the meantime you can access `text` property of each property setter and look for `field =` string. 

```kotlin
Konsist
            .scopeFromProject()
            .properties()
            .setters
            .assertTrue { it.text.contains("field =") }
```

**Reply by mahdibohloul** [reply]
Thanks for your attention. Do we have any timeline for this feature?
